### PR TITLE
Tpan/csharp original template

### PR DESCRIPTION
**Pull request by ti55987**

### CODE REVIEW

File: swagger-templates/csharp/ApiClient.mustache [hunk 1]
```diff
@@ -145,6 +154,10 @@ namespace {{packageName}}.Client
                 {
                     request.AddParameter(contentType, postBody, ParameterType.RequestBody);
                 }
+            } 
+            else 
+            {
```
Comment (DanGe42) [inline swagger-templates/csharp/ApiClient.mustache]: Is this for the empty request body case? Would it be necessary to add a comment explaining that our API explicitly requires an empty JSON body for requests that don't take parameters?

File: swagger-templates/csharp/ApiClient.mustache [hunk 2]
```diff
@@ -112,7 +112,16 @@ namespace {{packageName}}.Client
 
             // add header parameter, if any
             foreach(var param in headerParams)
-                request.AddHeader(param.Key, param.Value);
+            {
+                if (param.Key.Equals("Authorization"))
```
Comment (DanGe42) [inline swagger-templates/csharp/ApiClient.mustache]: Hmm, I'm not that familiar with C#, but I wonder if it's more conventional to use `string1 == string2`, or `string1.Equals(string2)`?

Comment (ti55987) [inline swagger-templates/csharp/ApiClient.mustache]: The Note in he ms documentation talking about string comparison
https://msdn.microsoft.com/en-us/library/cc165449.aspx

File: swagger-templates/csharp/ApiClient.mustache (hunk 1 above)
Comment (ti55987) [inline swagger-templates/csharp/ApiClient.mustache]: updated!

**Comment by DanGe42** [comment]
If the generated SDK works and has working acceptance tests, then this is good with me. I'm not super familiar with C# enough to have meaningful feedback.

**Comment by ti55987** [comment]
@DanGe42  Understood! But thank you very much for taking time on this PR!!!

**Pull request merged**